rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/OperatorPolicyReconciler.java

```diff
@@ -0,0 +1,81 @@
+        final RabbitMQConnectionInfo connectionInfo = new RabbitMQConnectionInfo(cluster.getName(), cluster.getNamespace(), RabbitMQServices.getDiscoveryServiceName(cluster.getName()));
+        final RabbitManagementApi apiClient = apiProvider.getApi(connectionInfo);
+
+        deleteObsoleteOperatorPolicies(cluster, apiClient);
+
+        for (final OperatorPolicySpec desiredPolicy : cluster.getOperatorPolicies()) {
+            final OperatorPolicy policy = new OperatorPolicy()
+                    .withName(desiredPolicy.getName())
+                    .withVhost(desiredPolicy.getVhost())
+                    .withApplyTo(OperatorPolicy.ApplyTo.fromValue(desiredPolicy.getApplyTo()))
+                    .withOperatorPolicyDefinition(desiredPolicy.getDefinition().asOperatorPolicyDefinition())
+                    .withPattern(Pattern.compile(desiredPolicy.getPattern()))
+                    .withPriority(desiredPolicy.getPriority());
+
+            try {
+                apiClient.createOperatorPolicy(policy.getVhost(), policy.getName(), policy);
+            } catch (final Exception e) {
+                log.error(String.format("Failed to create operator policy with name %s for cluster %s in namespace %s", policy.getName(), cluster.getName(), cluster.getNamespace()), e);
+            }
+        }
+    }
+
+    private void deleteObsoleteOperatorPolicies(final RabbitMQCluster cluster, final RabbitManagementApi apiClient) {
+        final String clusterName = cluster.getName();
+        final String clusterNamespace = cluster.getNamespace();
+
+        final List<OperatorPolicy> existingPolicies;
+
+        try {
+            existingPolicies = apiClient.listOperatorPolicies();
+        } catch (final Exception e) {
+            throw new RuntimeException(String.format("Unable to retrieve existing operator policies for cluster %s in namespace %s, skipping reconciliation of operator policies", clusterName, clusterNamespace), e);
+        }
+
+        final Map<String, OperatorPolicy> existingPolicyMap = existingPolicies.stream()
+                .collect(Collectors.toMap(OperatorPolicy::getName, policy -> policy));
+        final Map<String, OperatorPolicySpec> desiredPolicyMap = cluster.getOperatorPolicies().stream()
+                .collect(Collectors.toMap(OperatorPolicySpec::getName, policy -> policy));
+
+        for (final Map.Entry<String, OperatorPolicy> existingPolicy : existingPolicyMap.entrySet()) {
+            final String policyName = existingPolicy.getKey();
+            if (!desiredPolicyMap.containsKey(policyName)) {
+                try {
+                    apiClient.deleteOperatorPolicy(existingPolicy.getValue().getVhost(), policyName);
+                } catch (final Exception e) {
+                    log.error(String.format("Failed to delete operator policy with name %s with %s vhost in cluster %s in namespace %s", policyName, existingPolicy.getValue().getVhost(), clusterName, clusterNamespace), e);
```

Comment:
Do we have MDC filling in some of these values?

Reply:
i'm waiting for #30 and then i'll fix up any log messages here.

Reply:
Ah, didn't realize that was still open.  Sounds good to me.